src/components/common/Loading.tsx

```diff
@@ -27,8 +27,18 @@ const Loading: React.FC<LoadingProps> = ({ text, asOverlay }) => {
     zIndex: 10,
   } as React.CSSProperties;
 
+  const style = {
+    position: 'fixed',
+    height: '100vh',
+    width: '100vw',
+    top: '0',
+    left: '0',
+    justifyContent: 'center',
```

Comment:
line 32 - 36 is duplicate of those in `overlaySpinnerStyle`, so perhaps pass them in as props to the `VStack `?